From 538a29571368429df647ad5e046816b43a196e2b Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 01 Jul 2021 10:27:33 +0300
Subject: [PATCH] [SW-46517] greco: modify sw all reset mask

Currently, whenever hard reset is issued after LKD insmod,
ARC1 is not able to access SRAM to run preboot and the bootflow
of FW gets stuck. This issue is been debugged, till then as a
workaround PSOC and ARC's are also reset as part of hard reset.

This patch will be reverted once all reset issues related to
secureboot are resolved.

Change-Id: I4322f30044cde646958d909766ded116db17ec7b
---

diff --git a/arch/greco_arc/common.c b/arch/greco_arc/common.c
index c75cd0e..8a76d8b 100644
--- a/arch/greco_arc/common.c
+++ b/arch/greco_arc/common.c
@@ -420,6 +420,17 @@
 	regmap_writel(ARC_RST_VALUE, mmPSOC_ARC0_CFG_ARC_RST);
 }
 
+#define SW_ALL_RST_L_MSK		0xC00FF8FF
+#define SW_ALL_RST_M_MSK		0xFFFFFFFF
+#define SW_ALL_RST_H_MSK		0x7FFFFF
+
+static void sw_all_rst_init(void)
+{
+	regmap_writel(SW_ALL_RST_L_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_L);
+	regmap_writel(SW_ALL_RST_M_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_M);
+	regmap_writel(SW_ALL_RST_H_MSK, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_H);
+}
+
 void hl_plat_init(void)
 {
 	init_arc();
@@ -427,6 +438,8 @@
 	axi_enable_drain();
 
 	sram_scrambling_init();
+
+	sw_all_rst_init();
 }
 
 void spi_ssi_cs0_get(unsigned int *mask, unsigned int *data_offset,
diff --git a/arch/greco_arc_gnu/reset.c b/arch/greco_arc_gnu/reset.c
index cd9e761..70a8aa8 100644
--- a/arch/greco_arc_gnu/reset.c
+++ b/arch/greco_arc_gnu/reset.c
@@ -14,7 +14,7 @@
 
 #include "common.h"
 
-#define SW_ALL_RST_L_MSK		0xC00FF8F2
+#define SW_ALL_RST_L_MSK		0xC00FF8FF
 #define SW_ALL_RST_M_MSK		0xFFFFFFFF
 #define SW_ALL_RST_H_MSK		0x7FFFFF
 
